### PR TITLE
Test that julia can start if sys.so is deleted

### PR DESCRIPTION
**Pull request by timholy**
This can catch some bootstrap errors. See https://github.com/JuliaLang/julia/commit/47767573d2f9891cdc5db43fc2dcf98cb2503535#commitcomment-9240188.

### CODE REVIEW

**Comment by timholy** [comment]
Presumably this will fail travis until #9740 gets merged. I can restart then.

**Comment by kmsquire** [comment]
Isn't this #9740? :-)

**Comment by timholy** [comment]
StackOverflow

**Comment by timholy** [comment]
I'm quite puzzled that this seems to have passed on linux even before the merger of #9738. Any ideas?

Also, anyone know where `sys.so` ends up on OSX?

**Comment by tkelman** [comment]
> Also, anyone know where sys.so ends up on OSX?

Same place but `.dylib`, I think

**Comment by tkelman** [comment]
Pretty sure you'll need semicolons at the end of the `rm` lines. An easier version with less yelling at yml syntax would be `rm -f /tmp/julia/lib/julia/sys.so /tmp/julia/lib/julia/sys.dylib`

**Comment by timholy** [comment]
Thanks, Tony. We'll see if this works.

**Comment by vtjnash** [comment]
alternatively, perhaps we should add this to the standard tests, doing a `cp sys.ji local.ji && ./julia -J local.ji` to test?

**Comment by timholy** [comment]
Thanks @vtjnash, that's a much better way to do it. New version pushed.

**Comment by timholy** [comment]
BTW, I verified locally that this does catch the kind of problem it was designed to catch.

**Comment by timholy** [comment]
I got a worker termination on the `sets` test: https://travis-ci.org/JuliaLang/julia/jobs/46847690

File: Makefile [hunk 1]
```diff
@@ -438,6 +438,7 @@ test: release
 	@$(MAKE) $(QUIET_MAKE) -C test default
 
 testall: release
+	cp $(build_prefix)/lib/julia/sys.ji local.ji && ./julia -J local.ji -e 'true' && rm local.ji
```
Comment (tkelman) [inline Makefile]: This won't work on windows, where there's no `./julia` symlink. Use `$(JULIA_EXECUTABLE)`

**Pull request merged**